cmd/kyma/alpha/enable/module/module.go

```diff
@@ -0,0 +1,146 @@
+		kyma alpha enable module my-module -c alpha -n kyma-system -k default-kyma
+`,
+		RunE:    func(cmd *cobra.Command, args []string) error { return c.Run(cmd.Context(), args) },
+		Aliases: []string{"mod", "mods", "modules"},
+	}
+
+	cmd.Flags().DurationVarP(
+		&o.Timeout, "timeout", "t", 1*time.Minute, "Maximum time for the operation to enable a module.",
+	)
+	cmd.Flags().StringVarP(
+		&o.Channel, "channel", "c", "",
+		"The channel of the module to enable.",
+	)
+	cmd.Flags().StringVarP(
+		&o.Namespace, "namespace", "n", cli.KymaNamespaceDefault,
+		"The namespace of the Kyma to use. An empty namespace defaults to 'kyma-system'",
+	)
+	cmd.Flags().StringVarP(
+		&o.KymaName, "kyma-name", "k", cli.KymaNameDefault,
+		"The name of the Kyma to use. An empty name uses 'default-kyma'",
+	)
+	cmd.Flags().BoolVarP(&o.Wait, "wait", "w", false,
+		"Wait until the given Kyma resource is ready",
+	)
+
+	return cmd
+}
+
+func (cmd *command) Run(ctx context.Context, args []string) error {
+	if !cmd.opts.NonInteractive {
+		cli.AlphaWarn()
+	}
+
+	if len(args) != 1 {
+		return errors.New("you must pass one Kyma module name to enable it")
+	}
+	moduleName := args[0]
+
+	if err := cmd.opts.validateFlags(); err != nil {
+		return err
+	}
+
+	ctx, cancel := context.WithTimeout(ctx, cmd.opts.Timeout)
+	defer cancel()
+
+	return cmd.run(ctx, moduleName)
+}
+
+func (cmd *command) run(ctx context.Context, moduleName string) error {
+	start := time.Now()
+	if err := cmd.EnsureClusterAccess(ctx, cmd.opts.Timeout); err != nil {
+		return err
+	}
+
+	kyma := types.NamespacedName{Name: cmd.opts.KymaName, Namespace: cmd.opts.Namespace}
+	moduleInteractor := module.NewInteractor(cmd.K8s, kyma)
+	modules, err := moduleInteractor.Get(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to get modules: %w", err)
+	}
+
+	desiredModules, err := enableModule(modules, moduleName, cmd.opts.Channel)
+	if err != nil {
+		return fmt.Errorf("could not find module to enable: %w", err)
+	}
+
+	if len(modules) != len(desiredModules) {
+		if err = moduleInteractor.Update(ctx, desiredModules); err != nil {
+			return err
+		}
+		if cmd.opts.Wait {
+			if err = moduleInteractor.WaitForKymaReadiness(); err != nil {
+				return err
+			}
+		}
+	}
+
+	l := cli.NewLogger(cmd.opts.Verbose).Sugar()
+	l.Infof("enabling module took %s", time.Since(start))
+
+	return nil
+}
+
+func enableModule(modules []interface{}, name, channel string) ([]interface{}, error) {
+	for _, m := range modules {
+		mod, _ := m.(map[string]interface{})
+		moduleName, found := mod["name"]
+		if !found {
+			return nil, errors.New("invalid item in modules spec: name field missing")
+		}
+		if moduleName == name {
+			moduleChannel, cFound := mod["channel"]
+			if cFound && moduleChannel == channel {
+				// module already enabled
+				return modules, nil
+			}
+		}
+	}
+
+	newModule := make(map[string]interface{})
+	newModule["name"] = name
+	newModule["channel"] = channel
```

Comment:
After rebasing on latest main, then building and running

`kyma provision k3d`
`kyma alpha deploy`
`kubectl apply -f https://raw.githubusercontent.com/kyma-project/kyma/main/modules/alpha/moduletemplate-keda.yaml`
`kyma enable module keda`

I receive: Kyma.operator.kyma-project.io "default-kyma" is invalid: spec.modules[0].channel: Invalid value: "": spec.modules[0].channel in body should be at least 3 chars long

I Think you need to exclude this in case the channel name is not set. Also maybe provide better documentation to mention that if not specified the default channel will be used.

Reply:
Thanks for the find. What is the default channel? alpha?
Also kymas `spec.modules` items can consist only of `name:` field? `channel` is optional right?

Reply:
so from my view you do not need a channel at all. you can add the module and the channel of the kyma will be used. The problem is that you are using unstructured and you sent an empty channel, so leaving it out if its empty should be sufficient 👍